From ee2f59492e51379da3fb11760c8b2d1bec7ebe48 Mon Sep 17 00:00:00 2001
From: "kylepzak@projectinitiative.io" <kylepzak@projectinitiative.io>
Date: Tue, 19 Aug 2025 21:30:12 -0500
Subject: [PATCH 022/108] FROMGIT(6.18): arm64: dts: rockchip: rk3588s-rock-5a:
 Add green power LED

The Radxa ROCK 5A board includes a green power LED that is defined in
the vendor device tree but is missing from the upstream kernel DTS file.

This causes the LED to be uncontrollable from the operating system, as no
entry is created for it under /sys/class/leds.

This patch adds the missing node to the leds block, creating a
"green:power" device and allowing the LED to be controlled by the kernel.

Signed-off-by: Kyle Petryszak <kylepzak@projectinitiative.io>
---
 arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
index f894742b1ebe..f70b49d9361a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
@@ -52,6 +52,13 @@ leds {
 		pinctrl-names = "default";
 		pinctrl-0 = <&io_led>;
 
+		power-led {
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_POWER;
+			gpios = <&gpio3 RK_PC4 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "default-on";
+		};
+
 		io-led {
 			color = <LED_COLOR_ID_BLUE>;
 			function = LED_FUNCTION_STATUS;
-- 
2.34.1

